src/blocks/KeccakTranscript.sol

```diff
@@ -1027,6 +1027,46 @@ library KeccakTranscriptLib {
         return absorb(keccak, label, input);
     }
 
+    function absorb(
+        KeccakTranscript memory keccak,
+        uint8[] memory label,
+        uint256[] memory inputs_0,
+        uint256[] memory inputs_1,
+        uint256[] memory inputs_2
+    ) public pure returns (KeccakTranscript memory) {
+        uint8[] memory input = new uint8[](32 * inputs_0.length + 32 * inputs_1.length + 32 * inputs_2.length);
+
+        uint256 input_index = 0;
+        for (uint256 i = 0; i < inputs_0.length; i++) {
+            uint8[] memory input_bytes = scalarToBytes(inputs_0[i]);
+
+            for (uint256 j = 0; j < 32; j++) {
+                input[input_index] = input_bytes[j];
+                input_index++;
+            }
+        }
+
+        for (uint256 i = 0; i < inputs_1.length; i++) {
+            uint8[] memory input_bytes = scalarToBytes(inputs_1[i]);
+
+            for (uint256 j = 0; j < 32; j++) {
+                input[input_index] = input_bytes[j];
+                input_index++;
+            }
+        }
+
+        for (uint256 i = 0; i < inputs_2.length; i++) {
+            uint8[] memory input_bytes = scalarToBytes(inputs_2[i]);
+
+            for (uint256 j = 0; j < 32; j++) {
+                input[input_index] = input_bytes[j];
+                input_index++;
+            }
+        }
+
+        return absorb(keccak, label, input);
```

Comment:
And about here I wish we had inline functions, but [alas](https://github.com/ethereum/solidity/issues/12782)